pkg/controller/registry/resolver/resolver.go

```diff
@@ -218,6 +234,88 @@ func (resolver *MultiSourceResolver) resolveCRDDescription(sourceRefs []registry
 	return nil, ownerName, nil
 }
 
+// resolveRBACStepResources returns a list of step resources required to satisfy the RBAC requirements of the given CSV's InstallStrategy
+func resolveRBACStepResources(csv *v1alpha1.ClusterServiceVersion) ([]v1alpha1.StepResource, error) {
+	var rbacSteps []v1alpha1.StepResource
+
+	// User a StrategyResolver to
+	strategyResolver := install.StrategyResolver{}
+	strategy, err := strategyResolver.UnmarshalStrategy(csv.Spec.InstallStrategy)
+	if err != nil {
+		return nil, err
+	}
+
+	// Assume the strategy is for a deployment
+	strategyDetailsDeployment, ok := strategy.(*install.StrategyDetailsDeployment)
+	if !ok {
+		return nil, fmt.Errorf("could not assert strategy implementation as deployment for CSV %s", csv.GetName())
+	}
+
+	// Track created ServiceAccount StepResources
+	serviceaccounts := map[string]struct{}{}
+
+	// Resolve Permissions as StepResources
+	for i, permission := range strategyDetailsDeployment.Permissions {
+		// Create ServiceAccount if necessary
+		if _, ok := serviceaccounts[permission.ServiceAccountName]; !ok {
+			serviceAccount := &corev1.ServiceAccount{
+				TypeMeta: metav1.TypeMeta{
+					Kind: "ServiceAccount",
+				},
+			}
+			serviceAccount.SetName(permission.ServiceAccountName)
+			ownerutil.AddNonBlockingOwner(serviceAccount, csv)
+			step, err := v1alpha1.NewStepResourceFromObject(serviceAccount, serviceAccount.GetName())
+			if err != nil {
+				return nil, err
+			}
+			rbacSteps = append(rbacSteps, step)
+
+			// Mark that a StepResource has been resolved for this ServiceAccount
+			serviceaccounts[permission.ServiceAccountName] = struct{}{}
+		}
+
+		// Create Role
+		role := &rbac.Role{
+			TypeMeta: metav1.TypeMeta{
+				Kind: "Role",
+			},
+			Rules: permission.Rules,
+		}
+		ownerutil.AddNonBlockingOwner(role, csv)
+		role.SetName(fmt.Sprintf("%s-role-%d", csv.GetName(), i))
```

Comment:
don't we only need `-%d` if there are two permissions with the same service account? not for every role?

Reply:
Also maybe leave off `role` since that's in the kind?